packages/orbit-components/src/InputSelect/index.tsx

```diff
@@ -0,0 +1,374 @@
+            setActiveIdx(nextIdx);
+            setActiveDescendant(refs[nextIdx].current?.id);
+
+            if (dropdownRef && dropdownRef.current) {
+              dropdownRef.current.scrollTop = refs[nextIdx].current?.offsetTop;
+            }
+          }
+        }
+
+        if (ev.keyCode === KEY_CODE.ARROW_UP) {
+          if (activeIdx > 0) {
+            const isPrevDisabled = results[activeIdx - 1] && results[activeIdx - 1].disabled;
+            const prevIdx = isPrevDisabled ? activeIdx - 2 : activeIdx - 1;
+
+            setActiveIdx(prevIdx);
+            setActiveDescendant(refs[prevIdx].current?.id);
+
+            if (dropdownRef && dropdownRef.current) {
+              dropdownRef.current.scrollTop = refs[prevIdx].current?.offsetTop;
+            }
+          }
+        }
+      },
+      [results, activeIdx, isOpened, refs],
+    );
+
+    const input = (
+      <InputField
+        label={isLargeMobile && label}
+        onFocus={handleFocus}
+        onBlur={handleBlur}
+        onChange={handleOnChange}
+        id={inputId}
+        placeholder={placeholder}
+        autoFocus={!isLargeMobile}
+        role="combobox"
+        value={inputValue}
+        onKeyDown={ev => {
+          if (onKeyDown) onKeyDown(ev);
+          handleDropdownKey(ev);
+        }}
+        ariaHasPopup={isOpened}
+        ariaExpanded={isOpened}
+        ariaAutocomplete="list"
+        ariaActiveDescendant={activeDescendant}
+        ariaControls={isOpened ? dropdownId : undefined}
+        autoComplete="off"
+        ref={ref}
+        suffix={
+          String(inputValue).length > 1 && (
+            <StyledCloseButton
+              onClick={() => {
+                setIsOpened(false);
+                setInputValue("");
+              }}
+              $disabled={disabled}
+            >
+              <CloseCircle color="primary" ariaLabel="Clear" />
+            </StyledCloseButton>
+          )
+        }
+        {...props}
+      />
+    );
+
+    const dropdown = isOpened && (
+      <StyledDropdown
+        role="listbox"
+        id={dropdownId}
+        aria-labelledby={inputId}
+        $maxHeight={maxHeight}
+        $maxWidth={maxWidth}
+        ref={dropdownRef}
+      >
+        {results.map(({ disabled: optDisabled, title, description }, idx) => {
+          const optionId = randomId(title);
+          const isSelected = selectedIdx === idx;
+
+          return (
+            <StyledListChoiceWrapper key={optionId} $active={!optDisabled && activeIdx === idx}>
+              <ListChoice
+                disabled={optDisabled}
+                id={optionId}
+                onClick={ev => {
+                  ev.preventDefault();
+                  if (onOptionSelect) onOptionSelect(results[idx]);
+                  setInputValue(title);
+                  setActiveIdx(idx);
+                  setSelectedIdx(idx);
+                  if (isLargeMobile) setIsOpened(false);
```

Comment:
it's 4 state updates, but only one update. React batches multiple state updates into one, no? So in other words, does not mean how many state update functions we have on that onClick, it still will be counted as a single update. It will behave differently only with async actions